DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1

```diff
@@ -1,137 +1,1262 @@
-function Get-TargetResource
+data LocalizedData
+{
+    # culture="en-US"
+    ConvertFrom-StringData @'
+SetTargetResourceInstallwhatIfMessage=Trying to create AppPool "{0}".
+SetTargetResourceUnInstallwhatIfMessage=Trying to remove AppPool "{0}".
+AppPoolNotFoundError=The requested AppPool "{0}" is not found on the target machine.
+AppPoolDiscoveryFailureError=Failure to get the requested AppPool "{0}" information from the target machine.
+AppPoolCreationFailureError=Failure to successfully create the AppPool "{0}".
+AppPoolRemovalFailureError=Failure to successfully remove the AppPool "{0}".
+AppPoolUpdateFailureError=Failure to successfully update the properties for AppPool "{0}".
+AppPoolCompareFailureError=Failure to successfully compare properties for AppPool "{0}".
+AppPoolStateFailureError=Failure to successfully set the state of the AppPool {0}.
+'@
+}
+
+function Invoke-AppCmd
 {
-    [CmdletBinding()]
-    [OutputType([System.Collections.Hashtable])]
     param
     (
-        [parameter(Mandatory = $true)]
-        [System.String]
-        $Name
+        [string[]] $Arguments,
+        [string] $Path = "$env:SystemRoot\system32\inetsrv\appcmd.exe"
     )
 
-    $Ensure = 'Absent'
-    $State  = 'Stopped'
-
-    # Check if webadministration module is present or not
-    if(!(Get-Module -ListAvailable -Name WebAdministration))
+    if (Test-Path $Path)
     {
-        Throw 'Please ensure that WebAdministration module is installed.'
+        & $Path $Arguments
     }
-
-    # Need to import explicitly to run for IIS:\AppPools
-    # Setting verbose to false to avoid seeing all the imported command in
-    # DSC configuration verbose messages when configuration is run with -Verbose specified
-    Import-Module WebAdministration -Verbose:$false
-
-    $AppPool = Get-Item -Path IIS:\AppPools\* | ? {$_.name -eq $Name}
-
-    if($AppPool -ne $null)
+    else
     {
-        $Ensure = 'Present'
-        $State  = $AppPool.state
-    }
-
-    $returnValue = @{
-        Name   = $Name
-        Ensure = $Ensure
-        State  = $State
+        throw "$Path does not exist"
     }
-
-    return $returnValue
 }
 
-
-function Set-TargetResource
+function Invoke-AppCmdState
```

Comment:
I'm confused why this is needed, isn't this just doing the same thing as Invoke-AppCmd?